.github/workflows/portable_linux_package_matrix.yml

```diff
@@ -1,45 +1,76 @@
-      # Default behavior could be either:
-      #   A) generate a version or date -based tag for each release
-      #   B) append assets to 'mainline-snapshot' with a version or date suffix
-      #   C) overwrite assets on 'mainline-snapshot'
-      release_tag:
-        description: Publish files to this release tag
-        required: true
+        default: "dev"
+      package_suffix:
         type: string
-
-  workflow_call:
+  # Trigger manually (typically to test the workflow or manually build a release [candidate])
+  workflow_dispatch:
     inputs:
-      package_suffix:
+      build_type:
+        description: The type of build version to produce ("rc", or "dev")
         type: string
-        default: -ADHOCBUILD
-      # TODO: make this optional. See above.
-      release_tag:
-        description: Publish files to this release tag
-        required: true
+        default: "rc"
+      package_suffix:
         type: string
+  # Trigger on a schedule to build nightly release candidates.
+  schedule:
+    # Runs at 11:00 AM UTC, which is 3:00 AM PST (UTC-8)
+    - cron: '0 11 * * *'
 
 permissions:
   contents: read
 
-concurrency:
-  # A PR number if a pull request and otherwise the commit hash. This cancels
-  # queued and in-progress runs for the same PR (presubmit) or commit
-  # (postsubmit). The workflow name is prepended to avoid conflicts between
-  # different workflows.
-  group: ${{ github.workflow }}-${{ github.event.number || github.sha }}
-  cancel-in-progress: true
-
 jobs:
+  setup_metadata:
+    if: ${{ github.repository_owner == 'ROCm' || github.event_name != 'schedule' }}
+    runs-on: ubuntu-24.04
+    outputs:
+      version: ${{ steps.version.outputs.version }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - name: Setup Python
+        uses: actions/setup-python@8d9ed9ac5c53483de85588cdf95a591a75ab9f55 # v5.5.0
+        with:
+          python-version: 3.12
+
+      # Compute version suffix based on inputs (default to 'rc')
+      - name: Compute rc version suffix
+        if: ${{ inputs.build_type == 'rc' || inputs.build_type == '' }}
+        run: |
+          version_suffix="$(printf 'rc%(%Y%m%d)T')"
+          echo "version_suffix=${version_suffix}" >> $GITHUB_ENV
+
+      - name: Compute dev version suffix
+        if: ${{ inputs.build_type == 'dev' }}
+        run: |
+          version_suffix=".dev0+${{ github.sha }}"
+          echo "version_suffix=${version_suffix}" >> $GITHUB_ENV
+
+      - name: Compute version
+        id: version
+        run: |
+          base_version=$(jq -r '.["rocm-version"]' version.json)
+          echo "version=${base_version}${version_suffix}" >> $GITHUB_OUTPUT
+
   portable_linux_packages:
+    name: ${{ matrix.target_bundle.amdgpu_family }}::Build Portable Linux
+    runs-on: ${{ github.repository_owner == 'ROCm' && 'azure-linux-scale-rocm' ||  'ubuntu-24.04'  }}
```

Comment:
nit: extra space
```suggestion
    runs-on: ${{ github.repository_owner == 'ROCm' && 'azure-linux-scale-rocm' || 'ubuntu-24.04'  }}
```

Reply:
Thanks for the quick review and fetching this!